drivers/gpio/gpio_davinci.c

```
@@ -64,10 +62,24 @@ struct gpio_davinci_config {
 	const struct pinctrl_dev_config *pcfg;
 };
 
+const unsigned int offset_array[5] = {0x10, 0x38, 0x60, 0x88, 0xb0};
+#define MAX_REGS_BANK ARRAY_SIZE(offset_array)
+
+static struct gpio_davinci_regs *gpio_davinci_get_regs(const struct device *dev, uint8_t bank)
+{
+	if (bank < MAX_REGS_BANK) {
```

Add an assert or fail here instead of returning NULL and fail somewhere else.